### PR TITLE
allow multiple instances for multithreading

### PR DESCRIPTION
**Pull request by alcantara09**
depends on:

- [x] https://github.com/rock-gazebo/gui-vizkit3d_world/pull/5

### CODE REVIEW

**Comment by alcantara09** [comment]
I did the changes to adapt the new QApplication and to use multiple instances of vizkit3d_world to allow multithreading.

However, I was afraid of breaking the backward compatibility and I am not very happy with the solution.
@doudou, what is your opinion? Can you point me to a more elegant solution?

File: tasks/Task.cpp [hunk 1]
```diff
@@ -7,29 +7,29 @@ using namespace vizkit3d_world;
...
 
 Task::~Task() {
 }
 
-void Task::setupJointsPorts() {
-
-    if (vizkit3dWorld) {
+void Task::setupJointsPorts() 
+{
+    if(vizkit3dWorlds[0])
```
Comment (doudou) [inline tasks/Task.cpp]: `vizkit3dWorlds.empty()` or you access uninitialized memory if the vector is empty

While you are at it, change the `if ( ) { BIG } else { return }` into early return (`if (! ...) return;`)

File: tasks/Task.cpp [hunk 2]
```diff
@@ -108,16 +113,20 @@ bool Task::configureHook() {
...
+                                        _ignored_models.get(),
+                                        _width.get(),
+                                        _height.get(),
+                                        60,
+                                        0.01,
+                                        1000);
+
+        vizkit3dWorlds.push_back(vizkit3dWorld);                                
+    }
+    vizkit3dWorld = vizkit3dWorlds[0];
```
Comment (doudou) [inline tasks/Task.cpp:127]: Same here, this accesses uninitialized memory if `_widgets` is zero. Either check that `_widgets` is non-zero or check for `empty()`.

File: tasks/Task.cpp [hunk 3]
```diff
@@ -150,7 +159,9 @@ void Task::cleanupHook() {
     TaskBase::cleanupHook();
 
     releaseJointsPorts();
-    delete vizkit3dWorld;
-    vizkit3dWorld = NULL;
```
Comment (doudou) [inline tasks/Task.cpp:154]: You should not remove this line.

File: tasks/Task.cpp [hunk 4]
```diff
@@ -150,7 +159,9 @@ void Task::cleanupHook() {
...
+    for(auto vizkit3dWorld: vizkit3dWorlds)
+    {
+        delete vizkit3dWorld;
+    }
 }
```
Comment (doudou) [inline tasks/Task.cpp:166]: Don't forget to clear the vector.

**Pull request merged**